src/CGIHandler.cpp

```diff
@@ -0,0 +1,352 @@
+}
+
+/**
+ * @brief Sets up the environment variables for the CGI script.
+ *
+ * This function takes the environment variables stored in the member variable `m_env`,
+ * formats them as one string, and stores them in the provided `bufferEnv` vector. It then
+ * converts these strings to character pointers and stores them in the `envp` vector,
+ * which is used by the CGI script.  It also ensures that the last element
+ * of envp is a NULL pointer, as required by exec family functions.
+ *
+ * @param bufferEnv A reference to a vector of strings where the formatted environment
+ *                  variables are stored.
+ * @param envp A reference to a vector of character pointers where the addresses of the
+ *             formatted environment variables are stored. This vector is used
+ *             by the CGI script.
+ */
+void CGIHandler::setEnvp(std::vector<std::string>& bufferEnv, std::vector<char*>& envp) const
+{
+	for (std::map<std::string, std::string>::const_iterator citer = m_env.begin(); citer != m_env.end(); citer++) {
+		std::string tmp = citer->first + '=' + citer->second;
+		bufferEnv.push_back(tmp);
+	}
+	for (std::vector<std::string>::iterator iter = bufferEnv.begin(); iter != bufferEnv.end(); iter++)
+		envp.push_back(&(*iter).at(0));
+	envp.push_back(NULL);
+}
+
+/**
+ * @brief Sets up the argument vector (argv) for the CGI script execution.
+ *
+ * This function populates the provided argv vector with pointers to the
+ * strings in the bufferArgv vector. It also ensures that the last element
+ * of argv is a NULL pointer, as required by exec family functions.
+ *
+ * @param bufferArgv A reference to a vector of strings that is used
+ *                   to populate the argv vector.
+ * @param argv A reference to a vector of char pointers that is
+ *             populated with pointers to the strings in bufferArgv.
+ */
+void CGIHandler::setArgv(std::vector<std::string>& bufferArgv, std::vector<char*>& argv)
+{
+	bufferArgv.push_back(m_cgiPath);
+	bufferArgv.push_back(m_env["SCRIPT_FILENAME"]);
+	for (std::vector<std::string>::iterator iter = bufferArgv.begin(); iter != bufferArgv.end(); iter++)
+		argv.push_back(&(*iter).at(0));
+	argv.push_back(NULL);
+}
+
+// HELPER FUNCTIONS
+
+/**
+ * @brief Extracts the path information from a given path based on the CGI extension.
+ *
+ * This function searches for the CGI extension within the provided path. If the extension
+ * is found, it then looks for the first occurrence of a '/' character after the extension.
+ * The substring starting from this '/' character is considered the path information and is returned.
+ * If the CGI extension or the '/' character is not found, an empty string is returned.
+ *
+ * @param path The full path from which to extract the path information.
+ * @return A string containing the path information if found, otherwise an empty string.
+ */
+std::string CGIHandler::extractPathInfo(const std::string& path)
```

Comment:
Don't think it will find usage anywhere else than in the CGIHandler tbh. It is implicitly tested in the correct creation of the environment and the according test "Ctor". If it is required elsewhere in the code, we can still move it.

Reply:
Fair enough